networklibrary/src/main/java/com/eis/smsnetwork/SMSNetworkManager.java

```diff
@@ -83,8 +131,16 @@ public void removeResource(String key, RemoveResourceListener<String, FailReason
      * @author Marco Cognolato
      */
     @Override
-    public void invite(SMSPeer peer, InviteListener<SMSPeer, FailReason> inviteListener) {
-
+    public void invite(SMSPeer peer, InviteListener<SMSPeer, SMSFailReason> inviteListener) {
+        boolean hasSucceeded = false;
+        try {
+            CommandExecutor.execute(new SMSInvitePeer(peer));
+            hasSucceeded = true;
+        } catch (Exception e) {
+            Log.e(LOG_KEY, "There's been an error: " + e);
+            inviteListener.onInvitationNotSent(peer, SMSFailReason.MESSAGE_SEND_ERROR);
+        }
+        if (hasSucceeded) inviteListener.onInvitationSent(peer);
```

Comment:
Have you heard anything from Joe? How is he doing?

Reply:
Joe Mama?

Reply:
![My disappointment is immeasurable and my day is ruined](https://i.kym-cdn.com/entries/icons/mobile/000/025/543/eca.jpg)